scrapers/nd/events.py

```diff
@@ -1,114 +1,174 @@
+                "sub_com": item["sub_com"],
+            }
+            self.events[event_key][item_time] = []
+            self.events[event_key][item_time].append(agenda_item_details)
+
+        yield from self.create_events()
+
+    def create_events(self):
+        event_collection = []
+
+        for event in self.events.keys():
+            date, com, loc = re.search(r"(.+)\-(.+)\-(.+)", event).groups()
+
+            start_time = self.events[event]["event_start_time"]
+            date_time = f"{date} {start_time}"
+            date_object = dateutil.parser.parse(date_time)
+            date_with_offset = self._tz.localize(date_object)
+
+            event_obj = Event(
+                name=com,
+                location_name=loc,
+                description="Standing Committee Hearing",
+                start_date=date_with_offset,
+            )
 
-                    event.add_document(
-                        link_text, link.xpath("@href")[0], media_type="application/pdf"
+            for item_key in self.events[event]["item_keys"]:
+                agenda = self.events[event][item_key]
+                for item in agenda:
+                    time = datetime.datetime.strptime(item_key, "%H:%M:%S").strftime(
+                        "%I:%M %p"
                     )
+                    descr_with_time = f"[{time}]: {item['description']}"
+                    item_descr = event_obj.add_agenda_item(descr_with_time)
+                    if item["bill_name"]:
+                        item_descr.add_bill(item["bill_name"])
+                    if item["sub_com"]:
+                        item_descr.extras["sub_committee"] = item["sub_com"]
+
+            event_obj.add_source(self.url)
+
+            event_collection.append(event_obj)
+
+        return event_collection
+
+
+class BillNameScraper(HtmlPage):
+    def __init__(self, source):
+        self.source = source
+
+    def get_bill_name(self):
+        response = requests.get(self.source)
+        content = lxml.html.fromstring(response.content)
+        bill_name_tag = content.xpath(".//div[@id='content']//h3")[0]
+        bill_name = bill_name_tag.text_content().strip()
+        return bill_name
+
+
+class EventsTable(HtmlPage):
+    events_path = "/legend/committee/hearings/public-schedule/"
+    source = f"https://www.ndlegis.gov{events_path}"
+
+    def process_page(self):
+        date_span = self.root.xpath(".//h5[2]")[0].text_content()
+        start_year, end_year = re.search(r"(\d{4}).+(\d{4})", date_span).groups()
+
+        table_rows = self.root.xpath(".//tbody//tr")
+        agenda_items_list = []
+
+        for row_item in table_rows:
+            columns = row_item.xpath("./td")
+            columns_content = [x.text_content().strip() for x in columns]
+            if len(columns_content) == 7:
+                columns_content = columns_content[:-1]
+
+            bill, part_date, com, sub_com, loc, descr = columns_content
```

Comment:
Given the number of values we're pulling from this column, is it worth adding an example column in a comment for later reference?

Reply:
For sure. [Is this example column](https://github.com/openstates/openstates-scrapers/pull/4162/commits/037209511528934a8e11f7616c83939e636b489a#diff-fa16689e28573ece469e8dfea222521c243ef203a1c8c974d48d6b050b3ed308R126-R128) what you were imagining, or at least does it seem satisfactory for later reference?